lib/src/main/java/com/shopify/checkoutsheetkit/rpc/RPCRequest.kt

```diff
@@ -0,0 +1,306 @@
+ *
+ * @param P The type of parameters for this request
+ * @param R The type of response payload this request expects
+ */
+public abstract class RPCRequest<P : Any, R : Any>(
+    /**
+     * An identifier established by the client. If null, this is a notification.
+     */
+    public override val id: String?,
+    /**
+     * The parameters from the JSON-RPC request.
+     */
+    public open val params: P,
+    /**
+     * The serializer for the response type.
+     */
+    private val responseSerializer: KSerializer<R>
+) : RespondableEvent {
+
+    /**
+     * The JSON-RPC version. Must be exactly "2.0".
+     */
+    public open val jsonrpc: String = "2.0"
+
+    /**
+     * Weak reference to the WebView for sending responses.
+     */
+    public var webView: WeakReference<WebView>? = null
+
+    /**
+     * Whether this is a notification (no response expected).
+     */
+    public val isNotification: Boolean get() = id == null
+
+    private var hasResponded = false
+
+    @OptIn(ExperimentalSerializationApi::class)
+    public val json: Json = Json {
+        ignoreUnknownKeys = true
+        encodeDefaults = true
+        explicitNulls = false // Exclude null fields from JSON output
+    }
+
+    /**
+     * The RPC method name for this request type.
+     * Subclasses must override this to provide their method name.
+     */
+    public abstract val method: String
+
+    /**
+     * Respond to this request with the specified payload.
+     *
+     * @param payload The response payload
+     */
+    @OptIn(InternalSerializationApi::class)
+    public fun respondWith(payload: R) {
+        ShopifyCheckoutSheetKit.log.d(
+            "RPCRequest",
+            "respondWith called for method '$method' with id '$id'. " +
+                "webView: ${webView?.get()}, hasResponded: $hasResponded, isNotification: $isNotification"
+        )
+
+        when {
+            hasResponded -> logMultipleResponseAttempt()
+            isNotification -> logNotificationResponseAttempt()
+            else -> handleValidResponse(payload)
+        }
+    }
+
+    @OptIn(InternalSerializationApi::class)
+    private fun handleValidResponse(payload: R) {
+        try {
+            validate(payload)
+            hasResponded = true
+            sendSuccessResponse(payload)
+        } catch (e: Exception) {
+            ShopifyCheckoutSheetKit.log.e(
+                "RPCRequest",
+                "Validation failed for RPC request '$method' with id '$id': ${e.message}"
+            )
+            respondWithError("Validation failed: ${e.message}")
+        }
+    }
+
+    @OptIn(InternalSerializationApi::class)
+    private fun sendSuccessResponse(payload: R) {
+        webView?.get()?.let { webView ->
+            try {
+                val payloadJson: JsonElement = json.encodeToJsonElement(
+                    payload::class.serializer() as KSerializer<R>,
```

Comment:
hm the reflection seems a bit strange can you elaborate on what led to this? I also see you a have an instance of `responseSerializer` on the class